python/lib/eeg.py

```diff
@@ -816,4 +927,74 @@ def create_and_insert_archive(self, files_to_archive, archive_rel_name,
+        """
+        Create an archive with all annotations files associated to a specific recording
+
+        :param files_to_archive: tuple with the list of files to include in
+                                 the archive
+         :type files_to_archive: tuple
+        :param archive_rel_name: path to the archive relative to data_dir
+         :type archive_rel_name: str
+        :param eeg_file_id     : PhysiologicalFileID
+         :type eeg_file_id     : int
+        """
+
+        # load the Physiological object that will be used to insert the
+        # physiological archive into the database
+        physiological = Physiological(self.db, self.verbose)
+
+        # check if archive is on the filesystem
+        archive_full_path = self.data_dir + archive_rel_name
+        blake2            = None
+        if os.path.isfile(archive_full_path):
+            blake2 = blake2b(archive_full_path.encode('utf-8')).hexdigest()
+
+        # check if archive already inserted in database and matches the one
+        # on the filesystem using blake2b hash
+        query = "SELECT * " \
+                "FROM physiological_annotation_archive " \
+                "WHERE PhysiologicalFileID = %s"
+        results = self.db.pselect(query=query, args=(eeg_file_id,))
+        
+        if results and results[0]:
+            result = results[0]
+            if not blake2:
+                message = '\nERROR: no archive was found on the filesystem ' + \
+                          'while an entry was found in the database for '   + \
+                          'PhysiologicalFileID = ' + str(eeg_file_id)
+                print(message)
+                exit(lib.exitcode.MISSING_FILES)
+            elif result['Blake2bHash'] != blake2:
+                message = '\nERROR: blake2b hash of ' + archive_full_path     +\
+                          ' does not match the one stored in the database.'   +\
+                          '\nblake2b of ' + archive_full_path + ': ' + blake2 +\
+                          '\nblake2b in the database: ' + result['blake2b_hash']
+                print(message)
+                exit(lib.exitcode.CORRUPTED_FILE)
+            else:
+                return
+
+        # create the archive file
+        utilities.create_archive(files_to_archive, archive_rel_name, self.data_dir)
+
+        # insert the archive file in physiological_annotation_archive
+        blake2 = blake2b(archive_full_path.encode('utf-8')).hexdigest()
+        archive_info = {
+            'PhysiologicalFileID': eeg_file_id,
+            'Blake2bHash'        : blake2,
+            'FilePath'           : archive_rel_name
+        }
 
+        # insert the archive into the physiological_annotation_archive table
+        archive_fields = ()
+        archive_values = ()
+        for key, value in archive_info.items():
+            archive_fields = archive_fields + (key,)
+            archive_values = archive_values + (value,)
+        self.db.insert(
```

Comment:
the insert statement should be in a function in `lib/database_lib/physiologicalannotationarchive.py` file